### PR TITLE
Added restaurants cards with skeleton loading

### PR DESCRIPTION
**Pull request by mehulsatardekar**
Live Preview: https://dainty-strudel-f76e45.netlify.app/restaurants

**Files To Check 👍** 

src/Hooks/Restaurants/Restaurantlist.js
 src/components/UI/RestaurantsPage/Filters/RestaurantFilter.css
  src/components/UI/RestaurantsPage/Filters/RestaurantFilter.jsx
 src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx
 src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/Skeleton-Cards/Restaurant-skeleton-loader.css
  src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/Skeleton-Cards/Skeleton-Card.jsx
 src/components/UI/RestaurantsPage/Restaurants-Lists/RestaurantLists.css
 src/components/UI/RestaurantsPage/Restaurants-Lists/RestaurantLists.jsx
 src/components/UI/RestaurantsPage/Restaurants-Search/RestaurantsSearch.css
  src/components/UI/RestaurantsPage/Restaurants-Search/RestaurantsSearch.jsx
 src/components/UI/RestaurantsPage/Restaurants/Restaurants.css
 src/components/UI/RestaurantsPage/Restaurants/Restaurants.jsx
 src/contexts/Restaurants/RestaurantsContext.js

### CODE REVIEW

File: src/Hooks/Restaurants/Restaurantlist.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const RestaurantList = ()=>{
+    const [restaurants, setRestaurants] = useState(null);
+    const productFetch = async () => {
+      try {
+        const restaurantData = await fetch("/api/products");
+        const restaurantJson = await restaurantData.json();
+        setRestaurants(restaurantJson.products);
+      } catch (e) {
+        setRestaurants("error occured");
```
Comment (bhtibrewal) [inline src/Hooks/Restaurants/Restaurantlist.js:11]: You should show the error mesaage here

File: src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  useEffect(() => {
+    setTimeout(() => {
+      productFetch();
+    }, 2000);
+  }, [setRestaurants]);
+
+  return (
+    <>
+      {restaurants &&
+        restaurants.map((e) => {
```
Comment (bhtibrewal) [inline src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx]: Suggestion don't write e here instead something meaningful, and destucture it here so dont have to repeatedly write e.

File: src/components/UI/RestaurantsPage/Restaurants-Search/RestaurantsSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import "./RestaurantsSearch.css";
+
+function RestaurantsSearch({ title, description }) {
+  return (
+    <section className="food-search-container">
+      <div className="flex flex-justify-center flex-align-item-center flex-column font-white pt-3">
+        <span className="font-bold banner-head-txt pb-1">{title}</span>
```
Comment (bhtibrewal) [inline src/components/UI/RestaurantsPage/Restaurants-Search/RestaurantsSearch.jsx]: Suggestion: you should use p tag here

File: src/contexts/Restaurants/RestaurantsContext.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { createContext } from "react";
+import RestaurantListHook from "../../Hooks/Restaurants/Restaurantlist";
+const RestroHome = createContext();
+
+const RestaurantsData = ({children})=>{
+    const {restaurants,setRestaurants,productFetch} = RestaurantListHook();
```
Comment (bhtibrewal) [inline src/contexts/Restaurants/RestaurantsContext.js:6]: If you are using a custom hook here it should start with use

File: src/components/UI/RestaurantsPage/Restaurants-Search/RestaurantsSearch.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import "./RestaurantsSearch.css";
+
+function RestaurantsSearch({ title, description }) {
```
Comment (bhtibrewal) [inline src/components/UI/RestaurantsPage/Restaurants-Search/RestaurantsSearch.jsx]: Suggestion: use Es6

File: src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                  loading="lazy"
+                  className="ui-card-img img-restros"
+                />
+                <div className="card-badge">
+                  <span>{e.discount}</span>
+                </div>
+              </div>
+              <div className="flex flex-column pl-04 pt-1">
+                <span className="font-bold pb-09 w-100">{e.hotelName}</span>
+                <span className="font-md-light pb-02 w-100">
```
Comment (bhtibrewal) [inline src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx:37]: Suggestion: use sematic html

**Review by bhtibrewal — APPROVED** [review]
Nice work on naming , and nicely structuring of code and Ui also looks good, just one suggestion avoid to much unnecessary divs and use semantic html. You are good to merge after these minor fixes

File: src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import { Link } from "react-router-dom";
+import { useEffect,  useContext } from "react";
+import SkeletonCard from "./Skeleton-Cards/Skeleton-Card";
+
+import { RestroHome } from "../../../../../contexts/Restaurants/RestaurantsContext";
```
Comment (rahulyadav139) [inline src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx:5]: You can barrel importing/exporting here.

**Review by rahulyadav139 — APPROVED** [review]

File: src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx (hunk 7 above)
Comment (mehulsatardekar) [reply src/components/UI/RestaurantsPage/Restaurants-Lists/Restaurant-Card/RestaurantCard.jsx:5]: @rahulyadav139 yah will do that it's in my to do list of this project you can check this in issue section of project

**Pull request merged**